### ISSUE TITLE
RStudio 1.3 responds slower than 1.2

### ISSUE THREAD

**Issue by GegznaV** [rstudio/rstudio#6821]
### System details

    RStudio Edition : Desktop >
    RStudio Version : 1.3.957
    OS Version      : Windows 10
    R Version       : 4.0.0

### Steps to reproduce the problem

During the last few weeks, I am switching between RStudio 1.2.5042 and the newest preview version of RStudio 1.3.  In general, RStudio 1.3 works slower. Sometimes when I push Ctrl+Shift+F, it RStudio freezes for several even tens of seconds before the search window opens. When I open Global Options, I have to wait as RStudio also freezes for several seconds. 

The last such behavior was when I tried to **close** the update packages window (I pushed "Cancel") and it took about 10 seconds to close:
![image](https://user-images.githubusercontent.com/12725868/81215396-f3dd4680-8fe1-11ea-87a7-afe93d05cd8b.png)

I even managed to take a print screen of Task Manager:

![image](https://user-images.githubusercontent.com/12725868/81215693-5fbfaf00-8fe2-11ea-884c-809029d534f1.png)

It seems that **Qt web engine process** in RStudio used a lot of power.

And here how these metrics look like when RStudio does not do anything:
![image](https://user-images.githubusercontent.com/12725868/81216359-7581a400-8fe3-11ea-9e19-67295d2e000d.png)

I'm not sure how to reproduce the issue as sometimes It does not happen. I usually work in RStudio projects in which most parts of the text are not in English and there are many non-English letters (I'm not sure if it is related).

These problems newer happen when I switch to RStudio 1.2.

### Describe the problem in detail

In general, RStudio 1.3 works and  responds slower than RStudio 1.2.5044 

### Describe the behavior you expected

I expect RStudio 1.3 to work without lag as RStudio 1.2 does.

**Comment by R-Hannibal** [comment]
@GegznaV thank you so much for filing the ticket and for adding so much detail. 

We've had a couple of reports of performance issues with the 1.3 release, but haven't been able to consistently reproduce it internally thus far which makes it very difficult to address.

If it's convenient do you mind providing a link to the project you have been working in when the issue occurred (if multiples, please provide the one that seems to have the highest reproduction rate)? 

Also, you stated that the issue is occurring sporadically in 1.3, does not seem to occur at all in 1.2. Is your R version set to 4.0.0 when running both RStudio 1.2 and 1.3?

**Comment by GegznaV** [comment]
There is nothing special about the project: [U08 - Copy.zip](https://github.com/rstudio/rstudio/files/4652025/U08.-.Copy.zip)

Everything is the same (R, packages, etc.) except the version of RStudio (1.2 vs 1.3). Now I tried `1.3.958-3`. The problem repeats every time I install RStudio 1.3.

Can it be that user settings (and other files) of RStudio 1.2 and RStudio 1.3  (e.g., in `C:/Users/User/AppData/Local/RStudio` or in other places) interact and result in laggy behavior of RStudio?

**Comment by R-Hannibal** [comment]
I'm still not able to reproduce this issue locally, even after trying in the same project provided by @GegznaV and with the additional information.

@jmcphers do you have any thoughts on what could be causing this issue?

**Comment by GegznaV** [comment]
Could you write the list of folders where I should search for RStudio 1.2 and RStudio 1.3 settings? I will try to remove them and see if the problem persists.

**Comment by R-Hannibal** [comment]
@GegznaV - Not sure that resetting local settings will resolve the issue, but since we're really not certain at this point what IS causing the issue it could be worth a try:

[Resetting RStudio's Desktop State](https://support.rstudio.com/hc/en-us/articles/200534577-Resetting-RStudio-Desktop-s-State)

**Comment by R-Hannibal** [comment]
No response so assuming this issue has gone away. Closing for now.

@GegznaV - please let us know if this issue is still impacting you and we'll re-open and take a look.

**Comment by GegznaV** [comment]
@R-Hannibal, the situation got better when I installed officially released (not preview) 1.3.x versions of RStudio. From time to time it worked slow (e.g., by pressing <kbd>Ctrl+Shift+F</kbd> to open global search and replace, sometimes RStudio froze for half a minute), but not always. But yesterday I had a task in which I couldn't stand the slowness of RStudio. So I uninstalled RStudio 1.3 and installed 1.2 instead so that I could work normally (e.g., without having to wait several or more seconds until I can type several letters or scroll a line). The task I was doing yesterday included functions `gregexpr()` and `regmatches()` (I'm not sure, I'll try to create a reproducible examle, if the effect reproduces).

So the issue is not solved although it is marked as it is.

**Comment by R-Hannibal** [comment]
@GegznaV thanks for getting back to us. I am both frustrated and perplexed that you're still experiencing these slowdown/lag issues with the 1.3 release. We're in a tough spot with this issue currently due to our inability to reproduce it internally. That is not to say the issue your experiencing is invalid, we just haven't been able to pin it down with consistent reproduction or causation as of yet.

Please let us know if you are able to get a reproducible example and we will definitely take another look at this. I will continue to monitor this ticket for updates and try and respond as quickly as possible. Thanks for your help, and apologies this issue is proving to be so difficult to pin down.

**Comment by kevinushey** [comment]
I'd be curious to know if activating (or deactivating) software rendering as described in https://support.rstudio.com/hc/en-us/articles/360017886674-Troubleshooting-RStudio-Rendering-Errors makes a difference.

**Comment by GegznaV** [comment]
@kevinushey I tried all 3 rendering engine options (Auto-Detect, Software, and Desktop OpenGL). Then I tried to remove ticks from both "Use GPU ..." fields. **None** on them helped.

@R-Hannibal I found the situation in which RStudio **1.3**.1070 + R **4.0**.2 hangs every time I follow these steps:

1. I use this project [rstudio-hangs.zip](https://github.com/rstudio/rstudio/files/4975870/rstudio-hangs.zip)
2. Open it as a regular RStudio project.
3. Open file `run-this-code.R`
4. Select all the rows in the file and press **<kbd>Clrt</kbd>+<kbd>Enter</kbd>**
5. After the object `all_file_contents` in line 10 is printed, RStudio hangs (cursor is blinking, but functionality is totally frozen). Sometimes, I additionally tried to scroll the console or Editor window. The only way to exit this is to kill the RStudio-related processes via Task Manager.

(NOTE. This happens only with a certain combination of R and RStudio versions (1.3 + 4.0). If I change either to RStudio **1.2**.5042 or to R **3.6**.3, the issue is not present. The result appears in less than a second.

<details>
<summary>Session info</summary>

```r
> sessioninfo::session_info()
- Session info --------------------------------------------------------
 setting  value                       
 version  R version 4.0.2 (2020-06-22)
 os       Windows 10 x64              
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  English_United States.1252  
 ctype    English_United States.1252  
 tz       Europe/Helsinki             
 date     2020-07-25                  

- Packages ------------------------------------------------------------
 package     * version date       lib source                            
 assertthat    0.2.1   2019-03-21 [1] CRAN (R 4.0.0)                    
 cli           2.0.2   2020-02-28 [1] CRAN (R 4.0.0)                    
 crayon        1.3.4   2017-09-16 [1] CRAN (R 4.0.0)                    
 digest        0.6.25  2020-02-23 [1] CRAN (R 4.0.0)                    
 evaluate      0.14    2019-05-28 [1] CRAN (R 4.0.0)                    
 fansi         0.4.1   2020-01-08 [1] CRAN (R 4.0.0)                    
 glue          1.4.1   2020-05-13 [1] CRAN (R 4.0.0)                    
 htmltools     0.5.0   2020-06-16 [1] CRAN (R 4.0.1)                    
 knitr         1.29    2020-06-23 [1] CRAN (R 4.0.2)                    
 rlang         0.4.7   2020-07-09 [1] CRAN (R 4.0.2)                    
 rmarkdown     2.3.2   2020-07-20 [1] Github (rstudio/rmarkdown@ff1b279)
 rstudioapi    0.11    2020-02-07 [1] CRAN (R 4.0.0)                    
 sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 4.0.0)                    
 withr         2.2.0   2020-04-20 [1] CRAN (R 4.0.0)                    
 xfun          0.16    2020-07-24 [1] CRAN (R 4.0.2)                    
 yaml          2.2.1   2020-02-01 [1] CRAN (R 4.0.0) 
```

</details>

**Comment by kevinushey** [comment]
> After the object all_file_contents in line 10 is printed, RStudio hangs (cursor is blinking, but functionality is totally frozen). Sometimes, I additionally tried to scroll the console or Editor window. The only way to exit this is to kill the RStudio-related processes via Task Manager.

Does this imply the slowness you're experiencing is caused specifically by large amounts of Console output? Does clearing the console (e.g. via <kbd>Ctrl + L</kbd>) make a difference?

**Comment by GegznaV** [comment]
> Does this imply the slowness you're experiencing is caused specifically by large amounts of Console output?

And yes, and no. It seems that the issue is related to character output. E.g., `as.data.frame(dplyr::starwars)` does not cause any issues, but  `as.character(dplyr::starwars)` makes RStudio hang (sometimes nothing appears in the console).

>  Does clearing the console (e.g. via Ctrl + L) make a difference?

Ctrl + L (after several or more than ten seconds) allows exit from the state of totally frozen RStudio. But after that RStudio is still quite laggy (e.g., lettest appear later than I print them).

Were you able to reproduce the issue with my example?

**Comment by kevinushey** [comment]
Sorry for the late response -- unfortunately, I wasn't able to reproduce. 😞

**Comment by GegznaV** [comment]
I searched for all files on my PC, that were installed with RStudio 1.2 and with RStudio 1.3 (in local apps data, roaming, "~/.R" and elsewhere). I removed/uninstalled everything including RStudio itself. I lost the state of all my projects but **the issue is not present anymore**.  Unfortunately, I do not know where exactly the issue was. But I think it was related to the transition from RStudio 1.2 to 1.3. I faced similar instability issues when RStudio 1.2 was released an I was transitioning from RStudio 1.1 to 1.2.
